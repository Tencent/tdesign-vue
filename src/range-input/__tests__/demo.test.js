/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/src/range-input/_example/base.vue';
import popupVue from '@/src/range-input/_example/popup.vue';
import sizeVue from '@/src/range-input/_example/size.vue';
import statusVue from '@/src/range-input/_example/status.vue';

const mapper = {
  baseVue,
  popupVue,
  sizeVue,
  statusVue,
};

describe('RangeInput', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`RangeInput ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
