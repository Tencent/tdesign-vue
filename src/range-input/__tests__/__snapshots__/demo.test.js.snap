// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput RangeInput baseVue demo works fine 1`] = `
<div
  class="t-range-input"
>
  <div
    class="t-range-input__inner"
  >
    <div
      class="t-input__wrap t-range-input__inner-left"
    >
      <div
        class="t-input t-size-m t-is-default"
      >
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
    <div
      class="t-range-input__inner-separator"
    >
      -
    </div>
    <div
      class="t-input__wrap t-range-input__inner-right"
    >
      <div
        class="t-input t-size-m t-is-default"
      >
        <input
          autocomplete=""
          class="t-input__inner"
          placeholder="请输入"
          type="text"
          unselectable="off"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`RangeInput RangeInput popupVue demo works fine 1`] = `
<div
  class="t-range-input-popup"
>
  <div
    class="t-range-input"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RangeInput RangeInput sizeVue demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="t-range-input t-size-s"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="t-range-input"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="t-range-input t-size-l"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RangeInput RangeInput statusVue demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="t-range-input"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="t-range-input"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="t-range-input"
  >
    <div
      class="t-range-input__inner"
    >
      <div
        class="t-input__wrap t-range-input__inner-left"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
      <div
        class="t-range-input__inner-separator"
      >
        -
      </div>
      <div
        class="t-input__wrap t-range-input__inner-right"
      >
        <div
          class="t-input t-size-m t-is-default"
        >
          <input
            autocomplete=""
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            unselectable="off"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
